### PR TITLE
Fixes #62: Get posted data for PUT and PATCH.

### LINKED ISSUE #62: Get posted data for PUT and PATCH.

### PR DESCRIPTION
**Pull request by e0ipso**
#62

### CODE REVIEW

**Comment by e0ipso** [comment]
@amitaibu I came across this while working on the OAuth stuff.

**Comment by amitaibu** [comment]
Right, I wonder how the curl I did to test worked for me. Anyway probably worth a test.
I can ask someone in Gizra to do it if you don't have the energy for that ;)

‫ב-9 ביונ 2014, בשעה 23:36, ‏Mateu Aguiló Bosch notifications@github.com כתב/ה:‬

**Comment by yakoub** [comment]
I'll write a test for the PATCH and PUT in [RestfulHookMenuTestCase.test](https://github.com/Gizra/restful/blob/7.x-1.x/tests/RestfulHookMenuTestCase.test)

**Comment by e0ipso** [comment]
I'm on it too

**Comment by yakoub** [comment]
i did : git remote add mateu https://github.com/mateu-aguilo-bosch/restful.git
then : git checkout -b 62 mateu/62-posted-data-put-patch
so that i can collaborate with mateu .

but the branch i'm on has very different changes relative to branch 7.x-1.x than what the pull request says .
in particular it has different number of test files .

i ran all tests and got : 128 passes, 2 fails, 16 exceptions, and 7 debug messages .

logically there should be only one difference which is in RestfulHookMenuTestCase.test .

**Comment by e0ipso** [comment]
Green tests. Ready to merge.

File: plugins/restful/RestfulEntityBase.php [hunk 1]
```diff
@@ -1071,6 +1072,16 @@ public function getUrl($request = NULL, $options = array(), $keep_query = TRUE)
   }
 
   /**
+   * Helper function to remove the application generated request data.
+   *
+   * @param &array $request
+   *   The request array to be modified.
+   */
+  public static function cleanRequest(&$request) {
```
Comment (amitaibu) [inline plugins/restful/RestfulEntityBase.php:1080]: why static -- is it going to be called from outside the handler's context?

File: tests/RestfulUpdateEntityCurlTestCase.test [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+    $this->account = $this->drupalCreateUser(array(
+      'administer site configuration',
+      'administer nodes',
+      'edit own article content',
+      'edit any article content'
+    ));
+    $this->drupalLogin($this->account);
+
+    $this->httpauth_credentials = $this->account->name . ':' . $this->account->pass_raw;
```
Comment (amitaibu) [inline tests/RestfulUpdateEntityCurlTestCase.test:56]: cool!

**Comment by amitaibu** [comment]
Merged, thanks.

File: plugins/restful/RestfulEntityBase.php (hunk 1 above)
Comment (e0ipso) [inline plugins/restful/RestfulEntityBase.php:1080]: Potentially, yes. You may want to get the leftovers removed from a `$request` that you got from a setter. I think that having it as `public static` is harmless and _potentially_ a useful thing.

**Pull request closed without merging**